### ISSUE TITLE
method does not returns data.

### ISSUE THREAD

**Issue by bhavikcex** [nuxt/nuxt.js#791]
```
<script>

import axios from 'axios'

export default {
    data () {
        return {
            firstRecord: 1,
            count: 6,
            searchResult: this.loadSearchResult()
        }
    },
    methods: {
        loadSearchResult () {
            return axios.get(store.state.serviceUrl+ 'boxes?q=' + this.$route.query.stext + '&firstRecord=' + 1 + '&count=' + 9)
            .then((response) => {
                return response.data.response.data
            })
        }
    }
}
</script>
```

Why loadSearchResult  returns empty to caller?

**Comment by Kylart** [comment]
```javascript
...
.then((response) => {
    return response.data.response.data
})
...
```

Maybe is this`undefined` ? axios' data is only in the `response.data`.

If not, the uri seems a bit off
```javascript
store.state.serviceUrl+ 'boxes?q=' + this.$route.query.stext + '&firstRecord=' + 1 + '&count=' + 9
```

Shouldn't this be 
```javascript
this.$store.state.serviceUrl+ 'boxes?q=' + this.$route.query.stext + '&firstRecord=' + 1 + '&count=' + 9
```

**Comment by bhavikcex** [comment]
@Kylart  i did console.log it outputs data.

**Comment by Kylart** [comment]
Maybe [asyncData](https://nuxtjs.org/guide/async-data) would be of use here then ? 

```javascript
...
async asyncData () {
    const { data } = await axios.get(`store.state.serviceUrl+ 'boxes?q=' + this.$route.query.stext + '&firstRecord=' + 1 + '&count=' + 9`)
    return { 
      searchResult: data.response.data
    }
  }
...
```

**Comment by pi0** [comment]
[axios](https://github.com/nuxt/modules/tree/master/modules/axios) module for nuxt is now ready. You may give it a try ;)

**Comment by fortil** [comment]
As say @Kylart for this case is not necessary axios, only you should to use `searchResult: this.$store.state.....`

**Comment by pi0** [comment]
fixed by [v2.14.4](https://github.com/nuxt/nuxt.js/releases/tag/v2.14.4)
